### ISSUE TITLE
lua-dev config broken

### ISSUE THREAD

**Issue by meijieru** [LunarVim/LunarVim#3002]
### Problem description

The recent update of https://github.com/folke/lua-dev.nvim deprecate the `lspconfig` field, which breaks our customized lsp config.

### LunarVim version

rolling-2519e074

### Neovim version (>= 0.7)

NVIM v0.7.2

### Operating system/version

archlinux

### Affected language servers

sumneko_lua

### Actual behavior

For example, `vim`, `packer_plugins` as global for diagnostics don't take effect.

### support info

```console
 Use [q] or [Esc] to quit the window
 
 Language client log: /home/jmei/.cache/nvim/lsp.log
 Detected filetype:   lua
 
 2 client(s) attached to this buffer: 
 
 Client: sumneko_lua (id: 1, pid: 8170, bufnr: [1])
 	filetypes:       lua
 	autostart:       true
 	root directory:  /home/jmei/lib/LunarVim
 	cmd:             lua-language-server
 
 Client: null-ls (id: 2, pid: 5001, bufnr: [1])
 	filetypes:       cpp, lua
 	autostart:       false
 	root directory:  /home/jmei/lib/LunarVim
 	cmd:             nvim
 
 Configured servers list: sumneko_lua
```

**Comment by lvimuser** [comment]
Can you confirm it is fixed after the PR?

**Comment by meijieru** [comment]
Still suffers from this problem

**Comment by lvimuser** [comment]
> Still suffers from this problem

What's the output of `:lua =vim.lsp.get_client_by_id(1).config.settings` ?

**Comment by meijieru** [comment]
```
{
  Lua = {
    completion = {
      callSnippet = "Both",
      keywordSnippet = "Both"
    },
    diagnostics = {
      disable = { "unused-vararg", "different-requires" },
      enable = true,
      globals = { "describe", "it", "before_each", "after_each" }
    },
    runtime = {
      version = "LuaJIT"
    },
    telemetry = {
      enable = false
    },
    workspace = {
      library = {
        ["/home/jmei/.local/share/lunarvim/lvim/lua"] = true
      },
      maxPreload = 100000,
      preloadFileSize = 10000
    }
  },
  <metatable> = {
    __tostring = <function 1>
  }
}
```

**Comment by lvimuser** [comment]
Could you try changing `~/.local/share/lunarvim/lvim/lua/lvim/lsp/providers/sumneko_lua.lua` to
```lua
local dev_opts = {
  library = {
    vimruntime = true, -- runtime path
    types = true, -- full signature, docs and completion of vim.api, vim.treesitter, vim.lsp and others
    -- plugins = true, -- installed opt or start plugins in packpath
    -- you can also specify the list of plugins to make available as a workspace library
    plugins = { "plenary.nvim" },
  },
  override = nil, -- function(root_dir, options) end,
}

local lua_dev_loaded, lua_dev = pcall(require, "lua-dev")
if lua_dev_loaded then
  dev_opts = lua_dev.setup(dev_opts)
end

local opts = {
  settings = {
    Lua = {
      diagnostics = {
        globals = { "vim", "lvim", "packer_plugins" },
      },
      workspace = {
        library = {
          [require("lvim.utils").join_paths(get_runtime_dir(), "lvim", "lua")] = true,
        },
        maxPreload = 100000,
        preloadFileSize = 10000,
      },
    },
  },
}

return vim.tbl_deep_extend("force", dev_opts, opts)
```

**Comment by meijieru** [comment]
It doesn't work.

**Comment by ChristianChiarulli** [comment]
why do we have this builtin by default?

**Comment by lvimuser** [comment]
> why do we have this builtin by default?

It's extremely useful. It also should enable default completions for `lvim.` (but I think `[require("lvim.utils").join_paths(get_runtime_dir(), "lvim", "lua")] = true,` should be `[require("lvim.utils").join_paths(get_runtime_dir(), "lvim")] = true,`

**Comment by meijieru** [comment]
```
[require("lvim.utils").join_paths(get_runtime_dir(), "lvim")] = true,
```
Still have the problem.

**Comment by lvimuser** [comment]
Are you using nlspsettings? (do you have a `sumneko_lua.json`) file?

**Comment by meijieru** [comment]
Yep
```json
{
  "Lua.runtime.version": "LuaJIT",
  "Lua.diagnostics.enable": true,
  "Lua.diagnostics.globals": ["describe", "it", "before_each", "after_each"],
  "Lua.diagnostics.disable": ["unused-vararg", "different-requires"],
  "Lua.completion.callSnippet": "Both",
  "Lua.completion.keywordSnippet": "Both"
}
```

**Comment by lvimuser** [comment]
> Yep
> 
> ```json
> {
>   "Lua.runtime.version": "LuaJIT",
>   "Lua.diagnostics.enable": true,
>   "Lua.diagnostics.globals": ["describe", "it", "before_each", "after_each"],
>   "Lua.diagnostics.disable": ["unused-vararg", "different-requires"],
>   "Lua.completion.callSnippet": "Both",
>   "Lua.completion.keywordSnippet": "Both"
> }
> ```

Can you try without the file? (both rolling and what I've sent, if possible). `nlspsettings` might not be playing nice.

**Comment by meijieru** [comment]
Removing the JSON file fixes the `lvim` as global.
However, the lvim runtime seems not added correctly.

**Comment by kylo252** [comment]
@meijieru, could you please verify the fix in #3035

**Comment by meijieru** [comment]
Problem solved, thx!
